src/ring_theory/algebraic_independent.lean

```diff
@@ -0,0 +1,433 @@
+/-
+Copyright (c) 2021 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import ring_theory.adjoin.basic
+import linear_algebra.linear_independent
+import ring_theory.mv_polynomial.basic
+import data.mv_polynomial.supported
+import ring_theory.algebraic
+import data.mv_polynomial.equiv
+/-!
+# Algebraic Independence
+
+This file defines algebraic independence of a family of element of an `R` algebra
+
+## Main definitions
+
+* `algebraic_independent` - `algebraic_independent R v` states the family of vectors `v`
+  is algebraically independent over `R`, meaning that the canonical map out of the multivariable
+  polynomial ring is injective.
+
+* `algebraic_independent.repr` - The canonical map from the subalgebra generated by an
+  algebraic independent family into the polynomial ring.
+
+## References
+
+* [Stacks: Transcendence](https://stacks.math.columbia.edu/tag/030D)
+
+## TODO
+Prove that a ring is an algebraic extension of the subalgebra generated by a transcendence basis.
+
+## Tags
+transcendence basis, transcendence degree, transcendence
+
+-/
+noncomputable theory
+
+open function set subalgebra mv_polynomial algebra
+open_locale classical big_operators
+
+universes u v w x
+
+variables {ι : Type*} {ι' : Type*} (R : Type*) {K : Type*}
+variables {A : Type*} {A' A'' : Type*} {V : Type u} {V' : Type*}
+variables (v : ι → A)
+variables [comm_ring R] [comm_ring A] [comm_ring A'] [comm_ring A'']
+variables [algebra R A] [algebra R A'] [algebra R A'']
+variables {a b : R} {x y : A}
+
+/-- `algebraic_independent R v` states the family of vectors `v`
+  is algebraically independent over `R`, meaning that the canonical
+  map out of the multivariable polynomial ring is injective. -/
+def algebraic_independent : Prop :=
+injective (mv_polynomial.aeval v : mv_polynomial ι R →ₐ[R] A)
+
+variables {R} {v}
+
+theorem algebraic_independent_iff_ker_eq_bot : algebraic_independent R v ↔
+  (mv_polynomial.aeval v : mv_polynomial ι R →ₐ[R] A).to_ring_hom.ker = ⊥ :=
+ring_hom.injective_iff_ker_eq_bot _
+
+theorem algebraic_independent_iff : algebraic_independent R v ↔
```

Comment:
Could you please provide the `mp` version of this below, for easy dot notation?
Something like `algebraic_independent.eq_zero_of_aeval_eq_zero`.